concrete/models/messages.py

```diff
@@ -11,53 +11,73 @@ class MyClass(BaseModel):
     response = self.client.beta.chat.completions.parse(messages=messages, temperature=self.OPENAI_TEMPERATURE, **kwargs)
     message = response.choices[0].message
     message_formatted: MyClass = message.parsed
+    
+    
+# Allowed primitives
+https://platform.openai.com/docs/guides/structured-outputs/supported-schemas
+The following types are supported for Structured Outputs:
+String, Number, Boolean, Integer, Object, Array, Enum, anyOf
 """
 
 from typing import List, Optional
 
 from pydantic import Field
 
-from .base import ConcreteBaseModel, KombuMixin
+from .base import ConcreteModel, KombuMixin
 
+# Tracks all message types created as a sub class of Message
+# Keys are not type sensitive
+RESPONSE_REGISTRY = {}
```

Comment:
```suggestion
RESPONSE_REGISTRY: dict[str, "Message"] = {}
```
sorry to ask you to improve my code lmfao

Reply:
lolol, I think this was my code actually, I yoinked the registry idea from your tool registry though

Reply:
lol